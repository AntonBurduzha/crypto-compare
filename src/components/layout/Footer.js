import * as React from 'react';
import { Layout } from 'antd';
import type { Element } from 'react';

const Footer = (): Element<any> => {
	return (
		<Layout.Footer style={{ textAlign: 'center' }}>
			Crypto compare Â©2018 Created by Anton Burduzha
		</Layout.Footer>
	);
};

export default Footer;
