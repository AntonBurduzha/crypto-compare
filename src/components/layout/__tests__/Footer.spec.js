import React from 'react';
import { shallow, mount } from 'enzyme';
import { Layout } from 'antd';
import Footer from '../Footer';

describe('Footer component', () => {
  it('renders without crashing', () => {
    const wrap = shallow(<Footer/>);
    expect(wrap).toBeDefined();
  });

  it('has default text', () => {
    const wrap = mount(<Footer/>);
    expect(wrap.text()).toEqual('Crypto compare Â©2018 Created by Anton Burduzha');
  });

  it('matches snapshot', () => {
    const wrap = shallow(<Footer/>);
    expect(wrap).toMatchSnapshot();
  })
});
