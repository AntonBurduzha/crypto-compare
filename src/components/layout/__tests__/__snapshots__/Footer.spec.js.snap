// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Crypto compare ©2018 Created by Anton Burduzha",
      "style": Object {
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": "Crypto compare ©2018 Created by Anton Burduzha",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Crypto compare ©2018 Created by Anton Burduzha",
        "style": Object {
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": "Crypto compare ©2018 Created by Anton Burduzha",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
